pkg/executor/insert.go

```diff
@@ -478,6 +483,13 @@ func (e *InsertExec) doDupRowUpdate(ctx context.Context, handle kv.Handle, oldRo
 	if err != nil {
 		return err
 	}
+
+	if autoColIdx >= 0 && e.Ctx().GetSessionVars().StmtCtx.AffectedRows() > 0 {
```

Comment:
Why use `e.Ctx().GetSessionVars().StmtCtx.AffectedRows() > 0` ?

Reply:
Table tb with data:
pk  a   b
1   1   1

Execute this statement:
insert into tb (a, b) values (1, 2)  on duplicate key update b = 1

mysql_insert_id() should be 0 rather than 1, 
that is why we need check AffectedRows() > 0